legend-engine-language-pure-dsl-service/src/main/java/org/finos/legend/engine/language/pure/dsl/service/compiler/toPureGraph/HelperServiceBuilder.java

```diff
@@ -290,4 +291,21 @@ else if (params instanceof MultiExecutionParameters)
         }
         throw new UnsupportedOperationException("Unsupported service execution type '" + params.getClass().getSimpleName() + "'");
     }
+
+    private static boolean isTypeCompatibleForAssignment(String type, String assignmentType)
```

Comment:
this method doesn't seem specific to here. Don't we have another method that would check that an instance value is compatible with a type ?